jsm.go

```diff
@@ -1515,6 +1518,31 @@ func (jsc *js) StreamNames(opts ...JSOpt) <-chan string {
 	return ch
 }
 
+// StreamNameBySubject returns a stream name that matches the subject.
+func (jsc *js) StreamNameBySubject(subj string) (string, error) {
```

Comment:
I am not sure this is right. Since this becomes a context level API, should you pass the js options as a vararg? Then inside you would call getJSContextOpts(), etc... User should be able to override the wait and apiSubj() still need fixing to make sure it uses any user domain provided override (granted not part of this PR).

Reply:
While I'm not a huge fan of the concept of having 1 common option type for all interface methods, in this case I agree. It makes sense to allow user to at least pass context/wait and be consistent with the rest of the interface at the same time.

Reply:
agree. Updated the signature, and used `jsc.apiRequestWithContext` to at least use the context that can be passed, but still, it's misleading to give the user option to pass `Domain` and `ApiPrefix` and not use it.
That's consistent and I agree to have it, but it only adds up to what #1104 refers to.